src/server/utils/logging/logger.js

```diff
@@ -26,24 +26,33 @@ import baseConfig from './config/base';
 export function createLogger() {
   const useProductionConfig = !!(argv.logFormat === 'machine' || process.env.NODE_ENV !== 'development');
 
-  let transport;
+  let pinoConfig = baseConfig;
+  const transportStreams = [];
+
+  if (process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
+    transportStreams.push(createOtelTransport({
+      grpc: true,
+      console: process.env.NODE_ENV === 'development' && argv.logFormat === 'machine',
+    }));
+    pinoConfig = deepmerge(baseConfig, otelConfig);
+  } else if (useProductionConfig) {
+    pinoConfig = deepmerge(baseConfig, productionConfig);
+  }
 
-  if (!useProductionConfig && !process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
+  if (!useProductionConfig) {
     // eslint-disable-next-line global-require -- do not load development logger in production
-    transport = require('./config/development').default;
-  } else if (process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
-    transport = createOtelTransport();
+    transportStreams.push(require('./config/development').default);
+  }
+
+  let transport;
```

Comment:
what if `transportStreams` is an empty array?

Reply:
then we don't have a transport. it is not required

Reply:
if `transportStreams` is an empty array the variable is not used, and we passed undefined for the transport